File: ios/Classes/BluetoothLePeripheralPlugin.swift
import Flutter
import UIKit


public class BluetoothLePeripheralPlugin: NSObject, FlutterPlugin {
    
    private let bluetoothPeripheralManager: BluetoothPeripheralManager
    
    
    public override init() {
        bluetoothPeripheralManager = BluetoothPeripheralManager()
    }
    
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "bluetooth_le_peripheral", binaryMessenger: registrar.messenger())
    let instance = BluetoothLePeripheralPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
      let args = call.arguments as? Dictionary<String, Any>
    switch call.method {
    case "start":
        startAdvertising(payload: args?["payload"] as? String ?? "", result: result)
    case "stop":
        stopAdvertising(result: result)
    default:
      result(FlutterMethodNotImplemented)
    }
  
  }
    func startAdvertising(payload: String, result: @escaping FlutterResult){
        bluetoothPeripheralManager.start(payload: payload)
        result(nil)
    }
    func stopAdvertising(result: @escaping FlutterResult){
        bluetoothPeripheralManager.stop()
        result(nil)
    }
}

File: android/src/main/kotlin/com/abderrahmaneouali/bluetooth_le_peripheral/BluetoothLePeripheralPlugin.kt
package com.abderrahmaneouali.bluetooth_le_peripheral

import android.content.Context
import android.hardware.camera2.CameraExtensionSession.StillCaptureLatency
import android.os.Build
import androidx.annotation.NonNull
import androidx.annotation.RequiresApi

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** BluetoothLePeripheralPlugin */
class BluetoothLePeripheralPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel
  private var context : Context? = null

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "bluetooth_le_peripheral")
    channel.setMethodCallHandler(this)
    context = flutterPluginBinding.applicationContext
  }

  @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
  override fun onMethodCall(call: MethodCall, result: Result) {
    var bluetoothPeripheralManager = BluetoothPeripheralManager()
    var args = call.arguments as HashMap<String, Any>
    if (call.method == "start") {
      bluetoothPeripheralManager.start(context!!, args.get("payload") as String)
      result.success(null)
    } else if (call.method == "stop"){
      bluetoothPeripheralManager.stop()
      result.success(null)
    }
    else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
    context = null;
  }
}
